src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.FullGlobalizationData.Unix.cs

```diff
@@ -67,6 +48,26 @@ private static string GetUtcFullDisplayName(string timeZoneId, string standardDi
         }
 #pragma warning restore IDE0060
 
+        private static void GetStandardDisplayName(string timeZoneId, ref string? displayName)
+        {
+            // Determine the culture to use
+            CultureInfo uiCulture = CultureInfo.CurrentUICulture;
+            if (uiCulture.Name.Length == 0)
+                uiCulture = CultureInfo.GetCultureInfo(FallbackCultureName); // ICU doesn't work nicely with InvariantCulture
```

Comment:
this is repeated many times now. can you please move it to its own method?

Reply:
Made a private property that would be set for the instance, so these wouldn't need to be calculated multiple times.